app/Console/Commands/GenerateSitemap.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+namespace App\Console\Commands;
+
+use App\Models\Page;
+use Carbon\Carbon;
+use Illuminate\Console\Command;
+use Illuminate\Support\Facades\Route;
+
+class GenerateSitemap extends Command
+{
+    /**
+     * The console command name.
+     *
+     * @var string
+     */
+    protected $signature = 'seo:generate-sitemap';
+
+    /**
+     * The console command description.
+     *
+     * @var string
+     */
+    protected $description = 'Generate the sitemap.';
+
+    /**
+     * Execute the console command.
+     *
+     * @return mixed
+     */
+    public function handle()
+    {
+        $routes = ['/' => ['en' => 'en', 'asl' => 'asl', 'fr' => 'fr', 'lsq' => 'lsq']];
+        // once deployed to the server, files have the same modified date, use README as a default last modified date
+        $lastmod = ['default' => Carbon::createFromTimestamp(filemtime('./README.md'))->toISOString()];
+        foreach (Route::getRoutes()->get('GET') as $route) {
+            if ($route->named(config('seo.sitemap.patterns'))) {
+                $routeURI = $route->uri();
+                [$locale, $url] = explode('/', $routeURI, 2);
+                if (array_key_exists($url, $routes)) {
+                    $routes[$url][$locale] = $routeURI;
+                } else {
+                    $routes[$url] = [$locale => $routeURI];
+                    $routeName = explode('.', $route->getName());
+                    $page = Page::where('slug->en', '=', $routeName[count($routeName) - 1])->first();
+                    if ($page) {
+                        $lastmod[$url] = $page->updated_at->toISOString();
```

Comment:
This is interesting, and would work with the current pages we have. I guess the issue is that it's technically possible for there to be more Pages in the future. Perhaps the privacy policy would use this scheme or some other document. At the moment we have removed Creating pages from the filament admin panel, but it is still possible to make them programmatically. Also the route name and slug matching may not always be the case.

Another thing is that we are now running lookups for Pages for every route that matches the patterns.

Let me know what your thoughts are on these. Perhaps this is something we live with for now and file a ticket to further refine in the future, or maybe there are other improvements that can be made now.

Reply:
Of I forgot to add that if you keep the above you can probably use `Page::firstWhere` instead of `Page::where()->first()`

Reply:
Thanks @jobara, few thoughts in my mind after reading your comment. We are currently checking pages those are publicly available, those have `.about.` in their route names. Would there be more complex public pages later those require Page model? About the matching route name and slug, since we know that slug is programatically generated using title of specific Page model, we could use route name that follows the slug; if we were to take the approach of using config to tell which route names we want to ignore first and loop through later, we need to update that config value every time we add a page we want to generate sitemap for.

Currently, we do run lookups for Pages for every route, but only for the ones with 'en'. Perhaps I misunderstood your suggested approach of "looping through all `Page` models and getting their urls", but that seemed like more lookup  operations because looping through all Page happens per locale; could you explain in details about the approach please?